File: ase/misc/clr/clrcmp.js

var hndlr = {
  'CompareTo' : function(x) {
    print("CompareTo: x=" + x);
    return -1;
  }
};
var IComparable = CLR.Load("mscorlib", "System.IComparable");
var dprx = IComparable(hndlr);
dprx.CompareTo(30);

File: ase/misc/clr/clrcmp.lua

local hndlr = {
  CompareTo = function(x) 
    print("CompareTo: x=" .. x)
    return 1
  end,
  FOOBAR = function()
    print("FOOBAR called")
  end
}
print("C1");
local IComparable = CLR.Load("mscorlib", "System.IComparable")
print("C2");
local dprx = IComparable(hndlr)
print("C3");
dprx.CompareTo(30);
dprx.FOOBAR(30);
print("C4");
